### PR TITLE
Update Truffle

### PR DESCRIPTION
**Pull request by eregon**
Binary distributions work again, also it is causing problems as graal has a much newer truffle.

### CODE REVIEW

File: test/truffle/compiler/can-we-fold-yet.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 source test/truffle/common.sh.inc
 
 # tr works around https://github.com/jline/jline2/issues/282
-jt ruby --graal -J-Dgraal.TruffleIterativePartialEscape=true doc/samples/can-we-fold-yet.rb < test/truffle/compiler/can-we-fold-yet/input.txt | tr -d '\r' > actual.txt
+jt ruby --graal -J-Dgraal.TruffleIterativePartialEscape=true doc/samples/can-we-fold-yet.rb < test/truffle/compiler/can-we-fold-yet/input.txt > actual.txt
```
Comment (chrisseaton) [inline test/truffle/compiler/can-we-fold-yet.sh]: Should the comment above go as well?

**Comment by chrisseaton** [comment]
Let me know if you need help with the TCK.

File: test/truffle/compiler/can-we-fold-yet.sh (hunk 1 above)
Comment (eregon) [inline test/truffle/compiler/can-we-fold-yet.sh]: Indeed.

**Comment by eregon** [comment]
Yes, any help for the TCK is welcome (I didn't change anything and the TCK itself didn't change so this sounds like a regression to me). There is an issue opened for this as well.

File: test/truffle/compiler/can-we-fold-yet.sh (hunk 1 above)
Comment (eregon) [inline test/truffle/compiler/can-we-fold-yet.sh]: Done.

**Comment by eregon** [comment]
@chrisseaton Ready for a second review. Most of the TCK passes, except 4 tests which seem regressions in Truffle.

**Pull request merged**